src/Services/FHIR/FhirImmunizationService.php

```diff
@@ -0,0 +1,217 @@
+use OpenEMR\FHIR\R4\FHIRResource\FHIRBundle;
+use OpenEMR\FHIR\R4\FHIRResource\FHIRBundle\FHIRBundleLink;
+use OpenEMR\FHIR\R4\FHIRResource\FHIRImmunization\FHIRImmunizationEducation;
+use OpenEMR\FHIR\R4\FHIRResource\FHIRImmunization\FHIRImmunizationPerformer;
+use OpenEMR\FHIR\R4\PHPFHIRResponseParser;
+
+class FhirImmunizationService
+{
+    
+    private $id;
+    
+    public function __construct()
+    {
+    }
+    
+    public function setId($id)
+    {
+        $this->id = $id;
+    }
+    
+    public function getAll($search)
+    {      
+        $SQL = "SELECT id,
+                    patient_id,
+                    administered_date,
+                    cvx_code,
+                    manufacturer,
+                    lot_number,
+                    administered_by_id,
+                    administered_by,
+                    education_date,
+                    note,
+                    create_date,
+                    amount_administered,
+                    amount_administered_unit,
+                    expiration_date,
+                    route,
+                    administration_site,
+                    completion_status,
+                    refusal_reason 
+                    FROM Immunizations";
+
+        if (isset($search['patient']))
+            $SQL .= " WHERE patient_id = ?;";
+
+        $immunizationResults = sqlStatement($SQL, $search['patient']);
+        $results = array();
+        while ($row = sqlFetchArray($immunizationResults)) {
+            array_push($results, $row);
+        }
+        return $results;
+    }
+
+    public function getOne($id)
+    {
+        $SQL = "SELECT id,
+                    patient_id,
+                    administered_date,
+                    cvx_code,
+                    manufacturer,
+                    lot_number,
+                    administered_by_id,
+                    administered_by,
+                    education_date,
+                    note,
+                    create_date,
+                    amount_administered,
+                    amount_administered_unit,
+                    expiration_date,
+                    route,
+                    administration_site,
+                    completion_status,
+                    refusal_reason 
+                    FROM Immunizations
+                    WHERE id = ?";
+
+        $sqlResult = sqlStatement($SQL, $id);
+        $result = sqlFetchArray($sqlResult);
+
+        return $result;        
+    }
+
+    public function createImmunizationResource($id = '', $data = '', $encode = true)
+    {     
+        $status = new FHIRImmunizationStatusCodes();
+        $statusCoding = new FHIRCoding();
+        if (isset($data['completion_status'])) {
+            $statusCoding = new FHIRCoding();
+            $statusCoding->setSystem('http://hl7.org/fhir/event-status');
+            $statusCoding->setCode($data['completion_status']);
+            $statusCoding->setDisplay($data['completion_status']);
```

Comment:
For now I think we need to translate our values to match hl7 value sets.
i.e in-progress | on-hold | completed | entered-in-error | stopped